utils/s2n_safety.h

```diff
@@ -345,6 +345,15 @@ extern int s2n_constant_time_pkcs1_unpad_or_dont(uint8_t * dst, const uint8_t *
  */
 #define DEFER_CLEANUP(_thealloc, _thecleanup) \
    __attribute__((cleanup(_thecleanup))) _thealloc
+/**
+ * Often we want to free memory on an error, but not on a success.
+ * We do this by declaring a variable with DEFER_CLEANUP, then zeroing
+ * that variable after success to prevent DEFER_CLEANUP from accessing
+ * and freeing any memory it allocated.
+ *
+ * This pattern is not intuitive, so a named macro makes it more readable.
+ */
+#define ZERO_TO_DISABLE_DEFER_CLEANUP(_thealloc) memset(&_thealloc, 0, sizeof(_thealloc))
```

Comment:
I think it should include the "zero", or some other indication that the variable isn't useable after this point. Because it doesn't magically remove the cleanup attribute from the variable, it just makes the variable useless.

Reply:
fair enough